src/utilities/download_TROPOMI.py

```diff
@@ -15,186 +18,62 @@ def download_TROPOMI(startdate, enddate, Sat_datadir):
         Sat_datadir  [str]            : TROPOMI data directory for storing data
 
     """
-    # offline: 11/28/18 to end of 2020? + parts of 01/21
-    # s3://meeo-s5p/OFFL/L2__CH4___/2018/MM/DD/*.nc
-    # reprocessed: 04/30/18 to 11/28/18
-    # s3://meeo-s5p/RPRO/L2__CH4___/2018/MM/DD/*.nc
+    # offline: 07/26/2022 to present
+    # s3://meeo-s5p/OFFL/L2__CH4___/YYYY/MM/DD/*.nc
+    # reprocessed: 04/30/2018 to 07/25/2022
+    # s3://meeo-s5p/RPRO/L2__CH4___/YYYY/MM/DD/*.nc
     # --no-sign-request
-    start_str = str(startdate)
-    start_year = start_str[:4]
-    start_month = start_str[5:7]
-    start_day = start_str[8:10]
 
-    end_str = str(enddate)
-    end_year = end_str[:4]
-    end_month = end_str[5:7]
-    end_day = end_str[8:10]
+    # list of all dates between startdate and enddate excluding enddate
+    download_dates = np.arange(startdate, enddate, dtype="datetime64[D]")
 
+    # we create a bash script to download the data using the aws cli
+    # This is supposedly faster than downloading data with boto3 via python
     DATA_DOWNLOAD_SCRIPT = "./auto_generated_download_script.sh"
     cmd_prefix = "aws s3 sync --only-show-errors "
     remote_root = "s3://meeo-s5p/"
-    # access number of days in each month easily
-    month_days = [31, [28, 29], 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
+
     with open(DATA_DOWNLOAD_SCRIPT, "w") as f:
         print("#!/bin/bash\n", file=f)
         print("# This script was generated by jacobian.py\n", file=f)
```

Comment:
I'm not sure this is true (this function is never called in jacobian.py, only in run_imi.sh).